Assets/Scripts/Combat/Combantants/Combatant.cs

```diff
@@ -282,4 +283,19 @@ public void ShowBarsUI()
         HealthBarUIPanel.GetComponent<CanvasGroup>().alpha = 1f;
         HealthBarUIPanel.GetComponent<CanvasGroup>().blocksRaycasts = true;
     }
+
+    public void ResetHitpoints()
+    {
+        if(CurrentHealthPoints == 0 && !IsAlive)
```

Comment:
It looks like in both cases, you're setting `CurrentHP = MaxHP`. If the intention is to make sure the player is alive, you can also state `IsAlive = true`, and play the Idle animation. I would remove the conditions entirely

Reply:
Fixed! Removed the conditions entirely. However, now when this function is called, unity gives a weird warning. Everything works fine though.